src/normalize_axes.cpp

```diff
@@ -0,0 +1,116 @@
+#include <migraphx/operation.hpp>
+#include <migraphx/normalize_axes.hpp>
+#include <migraphx/stringutils.hpp>
+
+namespace migraphx {
+inline namespace MIGRAPHX_INLINE_NS {
+
+void normalize_axes(operation& op, const std::vector<std::size_t>& lens)
+{
+    int64_t n_dim = static_cast<int64_t>(lens.size());
+    auto val      = op.attributes();
+    if(val.contains("axis"))
+    {
+        auto axis = val["axis"].without_key().to<int64_t>();
+        if(op.name() == "flatten")
```

Comment:
I would like to avoid checking operator names here, but I think we need a richer set of attributes. We can leave it for now.

Reply:
How about the following attributes for each field:
1) is_array
2) clip_max (slice)
3) clip_min (slice)
4) include_max (flatten)
5) include_min
6) use_output_rank (unsqueeze)

Reply:
That sounds good, except we dont need an is_array attribute as we can infer that from the value.
Also, would this be in addition to the other attributes?

Reply:
For "Also, would this be in addition to the other attributes?", what does it mean for other attributes?